vermouth/molecule.py

```diff
@@ -996,6 +996,24 @@ def add_atom(self, atom):
             raise ValueError('Atom has no atomname: "{}".'.format(atom))
         self.add_node(name, **atom)
 
+
+    def add_atom_from_index(self, atom, index):
+        """
+        Sane as add_atom but use index name 
+
+        Parameters
+        ----------
+        atom: collections.abc.Mapping
+            The attributes of the atom to add. Must contain 'atomname'
+
+        Raises
+        ------
+        ValueError
+            If `atom` does not contain 'atomname'
+        """
+        self.add_node(index, **atom)
```

Comment:
I don't see the added value of this method.

Reply:
If you want to decouple the atom-name from the node name to use an index as node name this is the cleanest way to do. Otherwise it becomes a shit show of having to rename the node, atomnames, and or interactions. In fact there are a bunch of checks and functions that prevent this or that directly delete corresponding interactions.

Reply:
I don't see the added value over just calling block.add_node(index, **attrs)

Reply:
ahh yes that also works of course, but I thought this is more clear in comparison to the original parser. But I can remove it.